ibis/backends/snowflake/__init__.py

```diff
@@ -17,7 +17,7 @@
 from pathlib import Path
 from typing import TYPE_CHECKING, Any
 from urllib.parse import parse_qs, urlparse
-from urllib.request import urlretrieve
+from urllib.request import unquote_plus, urlretrieve
```

Comment:
I think most of the CI failures stem from this import, which should be from `urllib.parse`